### PR TITLE
DebugOptimzation with set Optimization flag to avoid overriding it from the build environment

### PR DESCRIPTION
**Pull request by HaseenaSainul**
In Yocto, RPI RDK build always used -Os optimisation level, this causing issue for Thunder debug build. 
So added option to set 0 optimisation level debug build to avoid this issues with different build system environment values

### CODE REVIEW

**Review by 404ag — DISMISSED** [review]
LGTM

**Review by pwielders — COMMENTED** [review]
I would expect that what you are now trying to achive is the BUILD_TYPE == "Debug". As this is called "Debug*Optimized*" I think it is correct that the O0 flag is not in there.
@bramoosterhuis  is tehre a "Debug" that does this ?

**Comment by HaseenaSainul** [comment]
I found this issue while enabling debubOptimized option for RPI yocto build.

**Comment by HaseenaSainul** [comment]
Same issue with Debug BuildType as well. So need to avoid the -Os level for debug* build types

File: cmake/modules/CompileSettingsDebug.cmake [hunk 1]
```diff
@@ -28,9 +28,14 @@ include(CMakePackageConfigHelpers)
...
-if("${BUILD_TYPE}" STREQUAL "DebugOptimized")
+if("${BUILD_TYPE}" STREQUAL "Debug")
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR ${CMAKE_COMPILER_IS_GNUCXX} )
-        target_compile_options(CompileSettingsDebug INTERFACE -g)
+        target_compile_options(CompileSettingsDebug INTERFACE -O0)
+    endif()
+
+elseif("${BUILD_TYPE}" STREQUAL "DebugOptimized")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR ${CMAKE_COMPILER_IS_GNUCXX} )
+        target_compile_options(CompileSettingsDebug INTERFACE -O1 -g)
```
Comment (pwielders) [inline cmake/modules/CompileSettingsDebug.cmake]: @bramoosterhuis / @HaseenaSainul just double checking: I would like to have the same optimization level in Release as in Debig so we test the same in both environments. I thought that the Release Optimisation level is 2 (-O2) if so, can we change it to -O2, if it is -O1 this pull request is good to go..

Comment (bramoosterhuis) [reply cmake/modules/CompileSettingsDebug.cmake]: I was hoping these flags would come from the build system. If we need to set them our selves I guess we want these flags:
DebugOptimized: "-O2 -g"
Debug: "-O0 -g"

Comment (HaseenaSainul) [reply cmake/modules/CompileSettingsDebug.cmake]: Yes,  but usage of -Os flag from build system is creating build issues. So think it is better to define the optimisation level from thunder itself. So update the flags as per your comments

**Review by pwielders — APPROVED** [review]

**Pull request merged**